UCLALibrary/libraryweb-site
# DRUP-1084: views more links

This PR fixes the broken "more" links on the News tab on Destinations and Locations:

Steps to review:

1) Visit a Location page, Eg. https://drup-1084-views-more-42glrty-sopc3ixpdigl4.us-2.platformsh.site/sel
2) Scroll down to the News block and click "More"
3) Ensure you're redirected to the /news page with the correct pre-filtered Location in the sidebar facets.


1) Visit a Destination page Eg https://drup-1084-views-more-42glrty-sopc3ixpdigl4.us-2.platformsh.site/destination/research-commons
2) Scroll down to the News block and click "More"
3) Ensure you're redirected to the /news page with the correct pre-filtered Location based on the destination in the sidebar facets.


Other things that were changed in this feature:

* The solution moves the more link out of the footer, from the views configuration screen, into a custom code solution in a preprocess function.
* This change caused a small issue with the theme's sass styles pulling in too many styles due to a poorly coded sass `@extends`, so I've refactored it to use a sass `%more-link` placeholder, instead of a normal classname. 
* The theme was also missing a `package-lock.json` file which was introduced a few years ago in npm version 5 (version 6 is out now).  This file lets us ensure everyone has the same npm package dependencies installed on their local machines.
